src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2752,6 +2767,9 @@ ConnectionStateMachine(WorkerSession *session)
 				}
 
 				PostgresPollingStatusType pollMode = PQconnectPoll(connection->pgConn);
+
+				HandleSocketChanges(session);
```

Comment:
shouldn't it be sufficient to get PQsocket before PQconnectPoll and then again after PQconnectPoll and compare?

Reply:
makes sense, it is simpler now